scripts/link-vms-by-sr.py

```diff
@@ -1,41 +1,37 @@
-#!/usr/bin/python
+#!/usr/bin/env python3
 # Populate a directory of symlinks partitioning VMs by SR
 # (c) Anil Madhavapeddy, Citrix Systems Inc, 2008
 
+from __future__ import print_function
+
+import argparse
 import atexit
+import contextlib
+import os
+import sys
+from pathlib import Path
+
 import XenAPI
-import os, sys
-import getopt
 
-def logout():
-    try:
+
+def logout(session):
+    with contextlib.suppress(Exception):
         session.xenapi.session.logout()
-    except:
-        pass
-atexit.register(logout)
```

Comment:
Where is the registration of the logout function happening now?

Reply:
The origin code is invalid, because the session is never passed in the `logout()` function and the trick is that the exception was catched.
Now I put the registration after session created https://github.com/xapi-project/xen-api/pull/5288/files#diff-12459e8d686a2efa25406c72b7fd866bb61c08c614c5030588fdff5304538499R77